### PR TITLE
[DQT] CBIGR/Biobank include redesigned DQT

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Includes latest redesigned DQT

### CODE REVIEW

**Comment by maltheism** [comment]
@HenriRabalais and @ridz1208 almost done with adding things needed for the redesigned DQT to the Loris branch cbigr is using. Please wait and i'll let you know when this PR is ready.

File: package.json [hunk 1]
```diff
@@ -11,27 +11,28 @@
     "@babel/plugin-transform-runtime": "^7.10.4",
```
Comment (ridz1208) [inline package.json:11]: @maltheism are the changes to this file necessary ?

File: project [hunk 2]
```diff
@@ -0,0 +1 @@
+../project_cbigr
```
Comment (ridz1208) [inline project]: @maltheism what is this file ?

File: src/Router/BaseRouter.php [hunk 3]
```diff
@@ -70,7 +70,7 @@ public function handle(ServerRequestInterface $request) : ResponseInterface
             if ($this->user instanceof \LORIS\AnonymousUser) {
```
Comment (ridz1208) [inline src/Router/BaseRouter.php]: @HenriRabalais do not acidentally merge this PR, this change in partiicular is only for cbigr open

File: webpack.config.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const config = [{
     './htdocs/js/components/Markdown.js': './jsx/Markdown.js',
```
Comment (ridz1208) [inline webpack.config.js:13]: move to project

File: project (hunk 2 above)
Comment (maltheism) [reply project]: @ridz1208 oh that's my project directory and for some reason it's there when it shouldn't be. I have a few project directories that I do `ln -s ../project_cbigr project` and so I can have multiple config.xml

File: package.json (hunk 1 above)
Comment (maltheism) [reply package.json:11]: @ridz1208 likely not. that happened when I was trying to get the new DQT to work. Turns out the htdocs/vender/js/react needed newer versions added.

File: modules/dqt/.gitignore [hunk 5]
```diff
@@ -0,0 +1,13 @@
+js/components
```
Comment (ridz1208) [inline modules/dqt/.gitignore]: @maltheism this file and all others inside `modules/dqt/...` should be copied over to the project git repo as an override on the module (the repository name is CBIGR) as well as the webpack content should be moved to webpack-project the entire module can simply be copy pasteed there

Comment (maltheism) [reply modules/dqt/.gitignore]: @ridz1208 I was having an issue getting @HenriRabalais's biobank module to work in the project directory. I was getting eslint error that I couldn't solve. So somewhat worried I might run into that while moving DQT to project but I'll try tomorrow.

**Comment by ridz1208** [comment]
@maltheism can you explain the changes in `/vendor/js` ?? they are too large to see

**Comment by ridz1208** [comment]
@maltheism @HenriRabalais The idea here is we want to avoid AS MUCH AS POSSIBLE changes on the loris end and instead have the changes on the project end to make the whole easier to maintain.

Some things are necessary such as the dashboard to dqt default module change but others might not be and lets remove them please

**Comment by maltheism** [comment]
@ridz1208 /vendor/js changes needed newer version of react and react-dom. They shouldn't break anything.

File: src/Router/BaseRouter.php (hunk 3 above)
Comment (ridz1208) [reply src/Router/BaseRouter.php]: actually, @maltheism could you undo this ? i'll make this change on another PR just to cbigr-open

**Comment by ridz1208** [comment]
Screenshot from 2020-10-16 10-16-43

visits not showing up

**Comment by maltheism** [comment]
@ridz1208 latest commit fixes visits not showing up.

**Comment by ridz1208** [comment]
Screenshot from 2020-10-19 17-04-14

@maltheism the select dropdown for visits doesnt close unless you re-click on the dropdown field. it would be nice if it just closes whereever you click on the page

**Comment by maltheism** [comment]
@ridz1208 I'll try to make that work. That issue with the dropdown also exists on the main branch of Loris.

File: php/libraries/Settings.class.inc [hunk 6]
```diff
@@ -61,7 +61,7 @@ class Settings
             return $url;
         }
         $www = $this->_config->getSetting('www');
-        return $www['url'];
+        return isset($www['url']) ? $www['url'] : '';
```
Comment (ridz1208) [inline php/libraries/Settings.class.inc]: whats this change for ?

File: src/Router/BaseRouter.php (hunk 3 above)
Comment (ridz1208) [reply src/Router/BaseRouter.php]: @maltheism cant merge this PR if you keep this

File: php/libraries/Settings.class.inc (hunk 6 above)
Comment (maltheism) [reply php/libraries/Settings.class.inc]: Loris won't work for my local setup without it. It just makes null become an empty string and so no error is thrown. @ridz1208

Comment (ridz1208) [reply php/libraries/Settings.class.inc]: it's an unecessary confusing change, you should find a way to fix it locally, or just remove it before submiting the PR please

Comment (maltheism) [reply php/libraries/Settings.class.inc]: @ridz1208 tried having something in the database where that url gets retrieved and it makes everything break. Unsure why but I can remove when everything is ready to be merged.

**Pull request merged**